### PR TITLE
Network - Local using channels

### PR DESCRIPTION
**Pull request by nikkolasg**
After all, if it makes Travis happy, why not using channels... I still prefer my `sync.Cond` solution but Travis is sad, so I'm sad. 
Let's see what Travis thinks of it.

### CODE REVIEW

File: network/local.go [hunk 1]
```diff
@@ -277,27 +276,27 @@ func (lc *LocalConn) Type() ConnType {
...
-	queue  [][]byte
+	queue  chan []byte
 	closed bool
+	sync.Mutex
 }
 
 func newConnQueue() *connQueue {
 	return &connQueue{
-		Cond: sync.NewCond(&sync.Mutex{}),
+		queue: make(chan []byte, 200),
```
Comment (ineiti) [inline network/local.go]: You would want to have that either as a constant or as a parameter.

File: network/local.go [hunk 2]
```diff
@@ -277,27 +276,27 @@ func (lc *LocalConn) Type() ConnType {
...
 	}
 }
 
 // push inserts a packet in the queue.
 // push won't work if the connQueue is already closed and silently return.
 func (c *connQueue) push(buff []byte) {
-	c.L.Lock()
-	defer c.L.Unlock()
+	c.Lock()
```
Comment (ineiti) [inline network/local.go]: I really think embedding mutexes is cool.

**Comment by ineiti** [comment]
I remember having thrown away even more code with regard to signalling that a new message is available. Did I dream that one up?

**Comment by nikkolasg** [comment]
Damn, and now a TLS error:
`fatal: unable to access 'https://gopkg.in/codegangsta/cli.v1/': gnutls_handshake() failed: A TLS fatal alert has been received.
`

**Comment by ineiti** [comment]
I think the tls-error is not ours... so a restart should be fine.

**Comment by ineiti** [comment]
Generally - do you want to push this now, or do you want to use this to find the error that still bugs us?

File: network/local.go (hunk 2 above)
Comment (Liamsi) [inline network/local.go]: Me too.

**Comment by nikkolasg** [comment]
Just to make things explicit: This PR passes all the time I've tried for the network/ tests now.
I'm just trying to debug some other datarace that occurs from time to time (and I think it's non related to this PR since it occurs at others PR also).

**Pull request closed without merging**